main/sceneManager.cpp

```diff
@@ -1,26 +1,66 @@
 #include "main/sceneManager.h"
 
-using Tri = objects::TriangleObj;
+SceneManager::SceneManager()
+{
+    createCollectors();
+}
+
+std::vector<objects::EnergyCollector *> SceneManager::getEnergyCollectors()
+{
+    std::vector<objects::EnergyCollector *> temp;
+    temp.reserve(_energyCollectors.size());
+    for (const auto &t : _energyCollectors)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
+
+std::vector<objects::TriangleObj *> SceneManager::getModelTriangles()
+{
+    std::vector<objects::TriangleObj *> temp;
+    temp.reserve(_modelTriangles.size());
+    for (const auto &t : _modelTriangles)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
 
-void IamSimulator()
+// PRIVATE METHODS
+
+void SceneManager::createCollectors()
 {
-    SceneManager manager;
-    CHECK(manager.loadModel("123231")) << "Cannot load model";
+    _energyCollectors.reserve(constants::kCollectors);
 
-    std::vector<Tri> triangles = manager.getModelTriangles();
-    generators::PointSource source;
+    const double collectorRadius = static_cast<double>(constants::kSimulationRadius) / 2;
+    double alphaIncrement;
 
-    for (int i = 0; i < 1000000; i++)
+    if (constants::kCollectors % 2 == 0)
+    {
+        alphaIncrement = 2 * constants::kPi * 180 / (static_cast<double>(constants::kCollectors) / 2 * 360);
+    }
+    else
     {
-        auto ray = source.generateRay(1, 2);
+        alphaIncrement = 2 * constants::kPi * 180 / ((static_cast<double>(constants::kCollectors) - 1) / 2 * 360);
+    }
 
-        for (const Tri &triangle : triangles)
+    for (double alpha = 0; alpha <= constants::kPi; alpha += alphaIncrement)
+    {
+        double cur = collectorRadius * std::sin(alpha);
+        double z = collectorRadius * std::cos(alpha);
+
+        core::Vec3 xAxesOrigin(cur, 0, z);
+        core::Vec3 yAxesOrigin(0, cur, z);
+
+        if (xAxesOrigin == core::Vec3(0, 0, constants::kSimulationRadius / 2))
```

Comment:
This is very convoluted. Why don't you just position 2 collectors at 0 and PI degrees?

Reply:
Also, now as I think about it, why do you add only one based on xAxesOrigin. Why does Y axis not have two collectors are the end?

Reply:
Ok, I think I know why you wanted to add only one collector.

But you messed up sin() and cos() functions...

Reply:
Because this position is only important when kCollectors is not even, and position Vec3(0, 0, kSimulationRadius) is somwhere in the middle in the vector<EnergyColelctors*>, so I did it this way. Maybe better idea would be to construct EnergyColletors 4 sides at once, and leave last element to be single if kColelctors is not even.

Reply:
> Ok, I think I know why you wanted to add only one collector (thanks to your comment on the method! so comments work! please use them...)
> 
> But you messed up sin() and cos() functions...

yes I did :P I will finish it later

Reply:
*fix it tommorow

Reply:
If this does not work, just drop it. Rewrite from scratch.